### ISSUE TITLE
hoc components require keys when displayed next to each other

### ISSUE THREAD

**Issue by zhipenglin** [storybooks/storybook#1814]
same hoc wrap component put in storybook

warning:Encountered two children with the same key, `1:$`. Child keys must be unique; when two children share a key, only the first child will be used.

such as:

``` javascript
const hoc=(WrapComponent)=>({...props})=><WrapComponent {...props} />

const Input=hoc(()=><input type="text" />);

const TextArea=hoc(({children})=><textarea>{children}</textarea>);

storiesOf('XXX', module)
    .add('XXX', withInfo()(() => <div>
        <Input/><TextArea></TextArea>
    </div>))
```

normal Running environment has not this warning.

**Comment by johnnyBira** [comment]
I have the same issue. Two different components using the same HOC. Is this related to storybook or react?
Feeling perplexed :/

**Comment by Hypnosphi** [comment]
Current workaround is to give your components different `displayName`s, like this:

const hoc=(WrapComponent)=>({...props})=><WrapComponent {...props} />

const Input=hoc(()=><input type="text" />);
Input.displayName = 'Input';

const TextArea=hoc(({children})=><textarea>{children}</textarea>);
TextArea.displayName = 'TextArea';

storiesOf('XXX', module)
    .add('XXX', withInfo()(() => <div>
        <Input/><TextArea></TextArea>
    </div>))

**Comment by johnnyBira** [comment]
Thank you.

Removed `addon-info` and not seeing the error any more. However the workaround dosen't do to for me.

```javascript
Column.displayName = 'Column';
export default getGridConfig(Column);
```

Still seeing the error and under info the HOC is displayed.

Instead of seeing:
```javascript
```

**Comment by Hypnosphi** [comment]
It should rather be
```
const GridColumn = getGridConfig(Column);
GridColumn.displayName = 'Column';
export default GridColumn;
```

There is also in issue regarding displaying HOCs in addon-info: https://github.com/storybooks/storybook/issues/938

**Comment by Hypnosphi** [comment]
By the way, the fix for key waning has landed in [3.2.15](https://github.com/storybooks/storybook/releases/tag/v3.2.15)
